pkg/hubble/metrics/http/handler.go

```diff
@@ -69,50 +74,106 @@ func (h *httpHandler) Status() string {
 	if h.context == nil {
 		return ""
 	}
-	return h.context.Status()
+	return h.context.Status() + fmt.Sprintf(",exemplars=%t", h.exemplars)
 }
 
 func (h *httpHandler) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error {
-	l7 := flow.GetL7()
+	if h.useV2 {
+		return h.processMetricsV2(flow)
+	} else {
+		return h.processMetricsV1(flow)
+	}
+}
+
+func (h *httpHandler) getMetadataFromFlow(flow *flowpb.Flow) (l7 *flowpb.Layer7, http *flowpb.HTTP, reporter string, traceID string, ok bool) {
```

Comment:
Named return parameters should be used wisely and I think this method returns too many values. For instance, why do we return `l7` and `http` when the calling methods already have a flow as parameter?

Reply:
I don't love it either, but it's just trying to get some code re-use. It's not a ton, but it's not a little either. If you have a concrete suggestion on how to improve this, I'm interested. The alternative I considered was returning a struct with these values, but that felt heavier.

Reply:
I'll just try breaking it up into a few small helpers and that should probably be enough re-use without a weird function like this.

Reply:
Refactored. PTAL.